### PR TITLE
Azure App Service support

### PR DESCRIPTION
**Pull request by jdubois**
Adds support to deploy applications to Azure App Service

- Works like the Azure Spring Cloud sub-generator for consistency
- Allow to deploy locally or using GitHub Actions
- Adds support for Azure Application Insights for monitoring

### CODE REVIEW

File: generators/azure-app-service/templates/application-azure.yml.ejs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# This configuration overrides the application.yml file.
+# ===================================================================
+
+# ===================================================================
+# Standard Spring Boot properties.
+# Full reference is available at:
+# http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html
+# ===================================================================
+
+<%_ if (azureAppInsightsInstrumentationKey != '') { _%>
```
Comment (MathieuAA) [inline generators/azure-app-service/templates/application-azure.yml.ejs]: !== (there should be a warning for this, amazing it never shows)

File: generators/generator-base.js [hunk 2]
```diff
@@ -1838,6 +1838,33 @@ module.exports = class extends PrivateBase {
         return child;
     }
 
+    /**
+     * run a command using the configured Java build tool.
+     *
+     * @param {String} buildTool - maven | gradle
+     * @param {String} profile - dev | prod
+     * @param {Boolean} command - the command (goal/task) to run
```
Comment (MathieuAA) [inline generators/generator-base.js]: a boolean? not a string?

File: generators/generator-base.js [hunk 3]
```diff
@@ -1838,6 +1838,33 @@ module.exports = class extends PrivateBase {
...
+     * @param {Function} cb - callback when build is complete
+     */
```
Comment (MathieuAA) [inline generators/generator-base.js]: there's a `@return` missing

File: generators/generator-base.js [hunk 4]
```diff
@@ -1838,6 +1838,33 @@ module.exports = class extends PrivateBase {
...
+        if (os.platform() !== 'win32') {
+            buildCmd = `./${buildCmd}`;
+        }
+        buildCmd += ` -P${profile}`;
+        this.log(`Running command: '${chalk.bold(buildCmd)}'`);
+        const child = {};
+        child.stdout = exec(buildCmd, { maxBuffer: 1024 * 10000 }, cb).stdout;
+        child.buildCmd = buildCmd;
+
+        return child;
```
Comment (MathieuAA) [inline generators/generator-base.js]: simpler:
```js
return {
  stdout: exec(buildCmd, { maxBuffer: 1024 * 10000 }, cb).stdout,
  buildCmd
}
```

File: generators/azure-app-service/templates/application-azure.yml.ejs (hunk 1 above)
Comment (jdubois) [reply generators/azure-app-service/templates/application-azure.yml.ejs]: Thanks!

File: generators/generator-base.js (hunk 2 above)
Comment (jdubois) [reply generators/generator-base.js]: Wrong copy/paste!! Thanks again

File: generators/generator-base.js (hunk 3 above)
Comment (jdubois) [reply generators/generator-base.js]: Same issue with the method above, I corrected both

File: generators/generator-base.js (hunk 4 above)
Comment (jdubois) [reply generators/generator-base.js]: Thanks for the tip! I'm also using this on the method above, as it was copy/pasted from there

File: generators/azure-app-service/templates/application-azure.yml.ejs (hunk 1 above)
Comment (clementdessoude) [reply generators/azure-app-service/templates/application-azure.yml.ejs]: FWIW, there is the eqeqeq rule in eslint which does that :)

Comment (jdubois) [reply generators/azure-app-service/templates/application-azure.yml.ejs]: OK but it does not seem to work on this file, or maybe we didn't configure it correctly

**Comment by MathieuAA** [comment]
Thanks @jdubois

**Pull request merged**